src/main/java/core/basesyntax/Wheel.java

```diff
@@ -1,5 +1,46 @@
 package core.basesyntax;
 
-public class Wheel {
+public class Wheel implements Cloneable {
     private int radius;
+
+    public Wheel(int radius) {
+        this.radius = radius;
+    }
+
+    public int getRadius() {
+        return radius;
+    }
+
+    public void setRadius(int radius) {
+        this.radius = radius;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        Wheel wheel = (Wheel) o;
+
+        return radius == wheel.radius;
+    }
+
+    @Override
+    public int hashCode() {
+        return radius;
+    }
+
+    @Override
+    protected Wheel clone() {
+        try {
+            return (Wheel) super.clone();
+        } catch (CloneNotSupportedException e) {
+            throw new RuntimeException("Can't create Wheel's clone");
```

Comment:
```suggestion
            throw new RuntimeException("Can't create Wheel's clone", e);
```
keep your exceptions consistent